### ISSUE TITLE
operator [] for flat_map always calls default constructor for mapped_type

### ISSUE THREAD

**Issue by r2r0** [ETLCPP/etl#692]
`operator []` calls default constructor for mapped_type even if flat_map has alrady element associated with given key.
Is it intended behaviour?
If key is already present in the map then I would expect that no new object is constructed.
For "heavy" objects it makes `operator []`  a little less usable.

**Comment by r2r0** [comment]
I did a small exmaple to make sure that my specific code does not influence result: [see here](https://godbolt.org/z/6vv9Ph7Pd)
And additionally I found something quite surprising for me - how many destructors (for mapped_type) are called during simple insertion.

**Comment by jwellbelove** [comment]
The changes I'm doing should reduce the number of time the mapped type is copied.

**Comment by jwellbelove** [comment]
With a bit of experimentation I've got it down to this...

```
**** Main
: S(1)
: S()
: operator =(S&&)
: ~S()
: myMap[1] set
: S(2)
: S()
: operator =(S&&)
: ~S()
: myMap[2] set
: myMap[2].val = 2
**** Main END
: ~S()
: ~S()
```
which is the same result I get from `std::map`.

**Comment by jwellbelove** [comment]
Fixed 20.36.0
